# Load data and Setup paths

In [18]:
from prepare_data.data_loader import get_data
from prepare_data.data_explorer import show_stats

# THE CORRECT PATH based on your file list:
path = "Data/raw/_annotations.coco.json"

# Load and Show
data, df_imgs, df_anns = get_data(path)
show_stats(df_imgs, df_anns, data)



Images: 500
Annotations: 959

--- Image Stats ---
Avg width: 1200.0
Avg height: 860.0

--- Categories ---
category_name
fire    959
Name: count, dtype: int64

--- Annotations per Image ---
Avg: 1.945233265720081
Max: 10
Min: 1


In [24]:
for cat in data['categories']:
    print(f"ID: {cat['id']} | Name: {cat['name']} | Supercategory: {cat.get('supercategory', 'None')}")

ID: 0 | Name: wildfire | Supercategory: none
ID: 1 | Name: fire | Supercategory: wildfire


"fire" belongs to "wildfire,"  essentially saying: "This is a fire, which is a type of wildfire."

Insights: 

Our initial analysis of the 500 satellite images reveals a total of 959 annotations. On average, each image contains 1.95 fire zones, with some images containing up to 10 distinct burned areas. This confirms that the model must be capable of multi-object detection within a single frame.

The Labeling Paradox:
While the dataset metadata defines two categories—"wildfire" (ID 0) and "fire" (ID 1)—our distribution analysis shows that 100% of the annotations (959/959) are tagged as "fire." The "wildfire" category exists only as a parent supercategory in the JSON structure and contains zero active labels.

Strategic Conclusion:
Given that we have a one-week development window, we have decided to simplify the project into a single-class detection task. Since "fire" is a sub-type of "wildfire," focusing exclusively on the "fire" label allows us to maximize model precision and avoid the technical debt of handling an empty class. This approach ensures a functional, high-performance model within the project's time constraints.